src/parser/datetime.rs

```diff
@@ -104,14 +194,112 @@ fn potential_interval_tokens(from: &DateTimeField) -> Vec<IntervalToken> {
     all_toks[offset..].to_vec()
 }
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+fn build_timezone_offset_second(
+    tokens: &Vec<IntervalToken>,
+    value: &str,
+) -> Result<i64, ParserError> {
+    use IntervalToken::*;
+    let all_formats = [
+        vec![Plus, Num(0), Colon, Num(0)],
+        vec![Dash, Num(0), Colon, Num(0)],
+        vec![Plus, Num(0), Num(0)],
+        vec![Dash, Num(0), Num(0)],
+        vec![Plus, Num(0)],
+        vec![Dash, Num(0)],
+        vec![Name("".to_string())],
+        vec![Zulu],
+    ];
+
+    let mut is_positive = true;
+    let mut hour_offset: i64 = -1;
+    let mut minute_offset: i64 = -1;
```

Comment:
this would be safer as `Option<i64>`, it would become impossible to return the value without checking or to use a value that has been checked incorrectly.

Reply:
done